elephant/conversion.py

```diff
@@ -152,3 +155,679 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+            spike_idx.append(l)
+        return spike_idx
+
+    def to_bool_array(self):
+        """
+        Returns a dense matrix (`scipy.sparse.csr_matrix`), which rows
+        represent the number of spike trains and the columns represent the
+        binned index position of a spike in a spike train.
+        The matrix columns contain only ones, which indicate a spike.
+
+        Returns
+        -------
+        bool matrix : numpy.ndarray
+            Returns a dense matrix representation of the sparse matrix,
+            with ones indicating **True** for a spike and **False*** for
+            non spike.
+            The **Trues** in the columns represent the index
+            position of the spike in the spike train and rows represent the
+            number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> import quantities as pq
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.TimeHistogram(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.to_bool_array())
+        [[ True  True False  True  True  True  True False False False]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+        """
+        return abs(scipy.sign(self.to_array())).astype(bool)
+
+    def to_array(self):
+        """
+        Returns a dense matrix, calculated from the sparse matrix with counted
+        time points, which rows represents the number of spike trains and the
+        columns represents the binned index position of a spike in a
+        spike train.
+        The  matrix columns contain the number of spikes that
+        occurred in the spike train(s).
+        If **bool** :attr:`store_mat` is set to **True** last calculated
+        matrix will be returned.
+
+        Returns
+        -------
+        matrix : numpy.ndarray
+            Matrix with spike times. Columns represent the index position of
+            the binned spike and rows represent the number of spike trains.
+
+        Examples
+        --------
+        >>> import elephant.conversion as conv
+        >>> import neo as n
+        >>> a = n.SpikeTrain([0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] * pq.s, t_stop=10.0 * pq.s)
+        >>> x = conv.TimeHistogram(a, num_bins=10, binsize=1 * pq.s, t_start=0 * pq.s)
+        >>> print(x.to_array())
+        [[2 1 0 1 1 1 1 0 0 0]]
+
+        See also
+        --------
+        scipy.sparse.csr_matrix
+        scipy.sparse.csr_matrix.toarray
+
+        """
+        if self._mat_u is not None:
+            return self._mat_u
+        if self._store_mat_u:
+            self._mat_u = self.to_sparse_array().toarray()
+            return self._mat_u
+        # Matrix on demand
+        else:
+            return self._sparse_mat_u.toarray()
+
+    def store_array(self):
```

Comment:
Yes it is new, but only the remove method. According to the discussion above https://github.com/NeuralEnsemble/elephant/pull/16#discussion-diff-23917521 I inserted it. Here a commit history would be benefical, as discussed yesterday ;) 

The argument `store_matrix` can be specified when instantiating the class, but is optional. If the user wants to store the matrix later on, there is still the possibility to do it with the method. With the remove method the user can explicitly remove the stored matrix.